lib/dry/cli/command.rb

```diff
@@ -13,37 +13,20 @@ class Command
       # @api private
       def self.inherited(base)
         super
+        base.class_eval do
+          @_mutex       = Mutex.new
+          @description  = nil
+          @examples     = []
+          @subcommands  = []
+          @arguments = base.superclass_variable_dup(:@arguments) || []
+          @options = base.superclass_variable_dup(:@options) || []
+        end
         base.extend ClassMethods
       end
 
       # @since 0.1.0
       # @api private
       module ClassMethods
-        # @since 0.1.0
-        # @api private
-        def self.extended(base)
-          super
-          base.class_eval do
-            @_mutex       = Mutex.new
-            @description  = nil
-            @examples     = []
-            @subcommands  = []
-            @arguments =
-              if base.superclass.instance_variable_defined?(:@arguments)
-                base.superclass.instance_variable_get(:@arguments).dup
-              else
-                []
-              end
-
-            @options =
-              if base.superclass.instance_variable_defined?(:@options)
-                base.superclass.instance_variable_get(:@options).dup
-              else
-                []
-              end
-          end
-        end
-
```

Comment:
This seems like an unrelated change or am I missing something?

Reply:
Yep, @jodosha asked to make this nicer after I merged previous PR

Reply:
@IvanShamatov OK in the future please use separate PRs with such unrelated improvements. Our future selves will appreciate it and it's a good habit to have in general. It makes maintenance simpler, especially when it turns out something needs to be reverted etc.

Reply:
will do that in separate pr